merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MissionModelGenerator.java

```diff
@@ -452,6 +512,8 @@ public JavaFile generateActivityActions(final MissionModelRecord missionModel) {
   public JavaFile generateActivityTypes(final MissionModelRecord missionModel) {
     final var typeName = missionModel.getTypesName();
 
+    // Unfortunately, JavaPoet 1.13.0 doesn't support generating Java 16 record definitions,
```

Comment:
Not sure if this would be worth our time, but we could make a wrapper method that would generate record syntax for us (and consider upstreaming it to JavaPoet 😉 )

Reply:
I figured it wouldn't be worth our time, but I *did* go hunting to see if there were any issues open on the subject. [There is one](https://github.com/square/javapoet/issues/829), and some of the branches referencing that issue seem to have taken a shot at this themselves. Nothing's been upstreamed yet though, and I didn't bother testing the capability in any of the forks.